setup.py

```diff
@@ -50,6 +50,7 @@
         'Products.CMFCore>=2.2.0dev',
         'Products.CMFDynamicViewFTI',  # required for cmf.ModifyViewTemplate
         'Products.CMFPlone',
+        'plone.app.contenttypes',  # required for the metadatafields vocabulary
```

Comment:
I am a little bit reluctant to make this dependency hard, because theoretically plone.app.content should be content type agnostic.
Anyway the improvement is much needed so at the end I am prone to merge it :)
Any opinion on this @mauritsvanrees @jensens?

Reply:
A little bit more work would be to move `plone.app.contenttypes.vocabulary` to `plone.app.vocabularies` and add a dependency on `plone.app.vocabularies` instead of `plone.app.contenttypes`. I would rename the vocabulary from `plone.app.contenttypes.metadatafields` to `plone.app.vocabularies.MetadataFields` and continue to register it with the name `plone.app.contenttypes.metadatafields` as well for backward compatibility.
What do you think?

Reply:
That would be the best, I think! 
I did not dare to ask because you already invested so much time in this, but this would be really cleaner.
I do not see it straightforward, because we have some risk of zcml collisions if the vocabulary is registered in both packages.
In order for that to be achieved we might need some zcml condition.

Another option is to register the vocabulary as `plone.app.vocabularies.metadatafields` and use something like:
```
voc_factory = queryUtility(IVocabularyFactory, 'plone.app.vocabulary.metadatafields')
if voc_factory is None:
    voc_factory = queryUtility(IVocabularyFactory, 'plone.app.contenttypes.metadatafields')
voc = voc_factory(self.context)
```
Then we should deprecate the `plone.app.contenttypes.metadatafields`.

If you have time to think this through it would be great, but as mentioned I am not requiring this because it might take quite some time to iron out the details.

Reply:
I don't see a reason for a conflict. You can register MetaDataFieldsVocabularyFactory with two vocabulary name in zcml.

Reply:
I mean if you have:
```
<utility name="plone.app.vocabularies.metadatafields" component="..." />
```
both in `plone.app.contenttypes` and `plone.app.vocabularies` you will have a conflict. But maybe this was not your intention. If not disregard my comment :)

Reply:
This is done. I moved it to https://github.com/plone/plone.app.vocabularies/pull/62